Document Datadog integration options for ClickHouse (#6419)
ClickHouse/clickhouse-docs: docs/cloud/features/08_monitoring/integrations.md

---

import CommunityMonitoring from '@site/docs/_snippets/_community_monitoring.md';
import PrivatePreviewBadge from '@theme/badges/PrivatePreviewBadge';

## Direct Datadog integration {#direct-datadog}
## Datadog integration {#direct-datadog}

Datadog offers a ClickHouse Monitoring plugin for its agent which queries system tables directly. This integration provides comprehensive database monitoring with cluster awareness through `clusterAllReplicas` functionality.
For Prometheus endpoint setup details, see the [Prometheus integration page](/integrations/prometheus#integrating-with-datadog).

> :::warning[Not recommended for ClickHouse Cloud]

Reviewer: Curious if we want to soften the warning here with the system tables approach, or if we want to keep this.

Author: I think it's ok personally without mentioning the system tables approach, but I'm indifferent.

|---|---|---|---|
| [ClickHouse Cloud Datadog API integration](#clickhouse-cloud-datadog-api-integration) *(recommended)* | [docs.datadoghq.com](https://docs.datadoghq.com/integrations/clickhouse-cloud/) | Cloud | Metrics |
| [ClickHouse Cloud Prometheus integration](#clickhouse-cloud-prometheus-integration) | [clickhouse.com/docs](https://clickhouse.com/docs/integrations/prometheus) | Cloud | Metrics |
| [ClickHouse Datadog Agent integration](#clickhouse-datadog-agent-integration) | [docs.datadoghq.com](https://docs.datadoghq.com/integrations/clickhouse/?tab=host) | OSS / Cloud | Logs, Metrics |
| [ClickHouse Datadog DBM](#clickhouse-datadog-dbm) | [datadoghq.com](https://www.datadoghq.com/product-preview/database-monitoring-for-clickhouse/) | OSS / Cloud | Logs, Metrics, Query Insights |

### ClickHouse Cloud Datadog API integration {#clickhouse-cloud-datadog-api-integration}

<PrivatePreviewBadge/>

The recommended way to serve service-level metrics from ClickHouse Cloud in Datadog.

> The customer provides a Datadog API key, and Datadog periodically polls the ClickHouse Cloud API to collect metrics using a push-based method authenticated via an OAuth handshake.

Reviewer: ```suggestion
You provide a Datadog API key, and Datadog periodically polls the ClickHouse Cloud API to collect metrics using a push-based method authenticated via an OAuth handshake.
```

Docs voice addresses the reader as "you" for a more informal, second person voice vs third person "the customer". There are a few instances of this throughout the doc.

- **Cloud only** — not suitable for OSS ClickHouse
- Metrics are **not** treated as custom metrics by Datadog
- Does **not** prevent the service from idling
- Ships with a pre-configured set of dashboards and monitors

> For onboarding steps, see [ClickHouse Cloud & Datadog - Integration](#).

Reviewer: ```suggestion
- Ships with a pre-configured set of dashboards and monitors

For onboarding steps, see [ClickHouse Cloud & Datadog - Integration](#).
```

Missing line break and missing link here

The customer configures the Datadog Agent with the [OpenMetrics integration](https://docs.datadoghq.com/integrations/openmetrics/) to periodically poll the ClickHouse Cloud API and collect metrics. See the [Prometheus integration page](/integrations/prometheus#integrating-with-datadog) for configuration details.

- **Cloud only** — not suitable for OSS ClickHouse
- Metrics are treated as **custom metrics** by Datadog
- Does **not** prevent the service from going idle

### ClickHouse Datadog Agent integration {#clickhouse-datadog-agent-integration}

The most common way to get data from on-premises ClickHouse services into Datadog. It also works with ClickHouse Cloud, with some caveats.

The Datadog Agent periodically polls the ClickHouse instance and collects metrics and logs. This integration ships with a pre-configured set of dashboards.

**OSS ClickHouse:** logs and metrics fully supported.

**ClickHouse Cloud (partial support):**
- Metrics only — logs are not supported
- Requires `datadog-cluster-agent` instead of `datadog-agent`
- **Will prevent the service from going idle**

### ClickHouse Datadog DBM {#clickhouse-datadog-dbm}

> Database Monitoring (DBM) for ClickHouse is developed and maintained  by Datadog. It works by configuring the Datadog Agent to collect system tables data from the ClickHouse instance, providing query-level performance insights similar to what the ClickHouse Cloud Console provides natively. ClickHouse has no involvement in this integration — issues should be directed to Datadog or resolved by the customer.

Reviewer: ```suggestion
Database Monitoring (DBM) for ClickHouse is developed and maintained by Datadog. It works by configuring the Datadog Agent to collect system tables data from the ClickHouse instance, providing query-level performance insights similar to what the ClickHouse Cloud Console provides natively. ClickHouse has no involvement in this integration — issues should be directed to Datadog or resolved by the customer.
```

> - Metrics are free; the customer pays extra for logs and system tables data ingestion

Reviewer: ```suggestion
- Metrics are free; you pay extra for logs and system tables data ingestion
```
